### ISSUE TITLE
[BUG][Python] Ranged response codes not working

### ISSUE THREAD

**Issue by Poolmann** [OpenAPITools/openapi-generator#14090]
#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

Ranged HTTP response codes are not recognized in deserialization of the response.

##### openapi-generator version

Version 6.3.0

##### OpenAPI declaration file content or url
With a response schema like:
```yaml
paths:
  /upload:
    post:
      tags: [upload]
      operationId: postUpload
      requestBody:
        $ref: "#/components/requestBodies/UploadRequest"
      responses:
        "2XX":
          $ref: "#/components/responses/UploadResponse"
        "403":
          $ref: "#/components/responses/UploadForbiddenError"
        "413":
          $ref: "#/components/responses/UploadRequestToLargeError"
        "415":
          $ref: "#/components/responses/UploadUnsupportedMediaError"
        "429":
          $ref: "#/components/responses/UploadTooManyRequestsError"
        # General responses
        "400":
          $ref: "#/components/responses/BadRequestError"
        "401":
          $ref: "#/components/responses/UnauthorizedError"
        "500":
          $ref: "#/components/responses/InternalServerError"
        "503":
          $ref: "#/components/responses/ServiceUnavailableError"
```

##### Generation Details

`java -jar openapi-generator-cli-6.3.0.jar generate -i openapi.yaml -g python`

##### Steps to reproduce

1. Use ranged response codes in the specification
2. skip_deserialization should be `False` (also default)

Expected Result: Deserialized response body
Actual Result: `<devaice_web_api_client.schemas.Unset object at 0x7fdf7fe177f0>`

These are the relevant code snippets regarding the deserialization:

```python showLineNumbers
if skip_deserialization:
    api_response = api_client.ApiResponseWithoutDeserialization(response=response)
else:
    response_for_status = _status_code_to_response.get(str(response.status))
    if response_for_status:
        api_response = response_for_status.deserialize(response, self.api_client.configuration)
    else:
        api_response = api_client.ApiResponseWithoutDeserialization(response=response)
```
and
```python
_status_code_to_response = {
    '403': _response_for_403,
    '413': _response_for_413,
    '415': _response_for_415,
    '429': _response_for_429,
    '400': _response_for_400,
    '401': _response_for_401,
    '500': _response_for_500,
    '503': _response_for_503,
    '2XX': _response_for_2XX,
}
```
So the response code '200' is not recognized => response_for_status is `None` and I don't get a deserialized response.

**Comment by spacether** [comment]
Thank you for reporting this. I didn't know about this feature. 
It was added in 
https://github.com/OAI/OpenAPI-Specification/pull/638/files

**Comment by spacether** [comment]
To get this working one will need to:
- first check if the response code is in the set of explicit codes like 202 404 etc
- then check if the response matches any defined wildcard responses, like 2XX 3XX
- then use a fallback default response if it exists

I am exposing the needed info:
- defaultResonse
- statusCodeResponses
- wildcardCodeResponses
In this PR: https://github.com/openapi-json-schema-tools/openapi-json-schema-generator/pull/97
which will make it easy to add this feature later.

**Comment by spacether** [comment]
FYI if you want to use this feature, it is now available in the v2.0.0 release of openapi-json-schem-generator
which is where development of the python generator is continuing.
See a working test of ranged responses here:
https://github.com/openapi-json-schema-tools/openapi-json-schema-generator/blob/master/samples/openapi3/client/petstore/python/tests_manual/test_paths/test_fake_wild_card_responses/test_get.py#L58
